src/Catalyst.Protocol/Wire/TransactionBroadcast.cs

```diff
@@ -42,6 +44,59 @@ public partial class TransactionBroadcast
              && ConfidentialEntries.All(e => e.IsValid());
         }
 
+        /// <summary>
+        /// TODO: discuss OnConstruction - why is it designed this way 
+        /// </summary>
+        public void AfterConstruction()
+        {
+            IsContractDeployment = ContractEntries.Any(c => c.IsValidDeploymentEntry);
+            IsContractCall = ContractEntries.Any(c => c.IsValidCallEntry);
+            IsPublicTransaction = PublicEntries.Any() && PublicEntries.All(e => e.IsValid());
+            IsConfidentialTransaction = ConfidentialEntries.Any()
+             && ConfidentialEntries.All(e => e.IsValid());
+        }
+
+        // TODO: discuss why TransactionBroadcastTest is commented out
+        public UInt256 AverageGasPrice
+        {
+            get
+            {
+                BigInteger averagePrice = BigInteger.Zero;
+                ulong totalLimit = 0;
+                int count = ContractEntries.Count;
+                for (int i = 0; i < count; i++)
+                {
+                    ulong limit = ContractEntries[i].GasLimit;
+                    totalLimit += limit;
+                    averagePrice += limit * ContractEntries[i].GasPrice;
+                }
+
+                if (totalLimit == 0)
+                {
+                    return UInt256.Zero;
+                }
+                
+                UInt256.Create(out UInt256 result, averagePrice / totalLimit);
+                return result;
+            }
+        }
+        
+        // TODO: discuss why TransactionBroadcastTest is commented out
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4350015)